app/validators/digital_object/projects_validator.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class DigitalObject::ProjectsValidator < ActiveModel::Validator
+  def validate(digital_object)
+    primary_string_key = digital_object.primary_project&.string_key
+    if digital_object.other_projects&.detect { |proj| proj.string_key == primary_string_key }
+      digital_object.errors.add(:other_projects, "Other projects cannot also be primary: #{primary_string_key}")
+      return false
+    end
+    true
+  end
```

Comment:
```suggestion
  def validate(digital_object)
    primary_string_key = digital_object.primary_project&.string_key
    if digital_object.other_projects&.detect { |proj| proj.string_key == primary_string_key }
      digital_object.errors.add(:other_projects, "Other projects cannot also be primary: #{primary_string_key}")
      return
    end
  end
```
Looks like we don't need to return true or false from the validate method.  Adding or not adding errors is all that's required: https://api.rubyonrails.org/classes/ActiveModel/Validator.html